compiler/rustc_middle/src/traits/solve/cache.rs

```diff
@@ -48,7 +42,7 @@ impl<'tcx> EvaluationCache<'tcx> {
         if cfg!(debug_assertions) {
             drop(map);
             let expected = CacheData { result, proof_tree, additional_depth, encountered_overflow };
-            let actual = self.get(tcx, key, [], Limit(additional_depth));
```

Comment:
`Limit` is defined in `rustc_session`, and I was *kinda* too lazy to find a good ancestor crate to put `Limit` in...

Could fix it as a follow-up or fix it in this PR if necessary.